pkg/cri/resource-manager/policy/builtin/memtier/node.go

```diff
@@ -403,6 +407,83 @@ func (n *node) GetMemoryType() memoryType {
 	return memoryMask
 }
 
+// MoveAllWorkloadsUp transferes all containers from this node to its
+// parent node.
+func (n *node) MoveAllWorkloadsUp() (bool, error) {
```

Comment:
We should not really move containers up. The reason is that we should not be reallocating their cpusets here. This could be called instead something like 'RelaxMemoryPinning' or 'ExpandMemoryPinning'.

Instead, what we are doing/should be doing is that once a (tree) node is in a situation where the guarantee of the memory requests might be violated (because of containers spread memory-wise across multiple (tree) nodes, we broaden cpuset.mems of containers 'up in the tree', one level at a time until the requests are guaranteed.

It is important to not reallocate cpusets: it is the only indirect control (by misusing our faint knowledge of how the kernel memory allocator works) to achieve that the container will keep allocating from its originally assigned 'primary' NUMA nodes, until those run out of memory.